internal/provider/tag_resource.go

```diff
@@ -0,0 +1,192 @@
+package provider
+
+import (
+	"context"
+	"errors"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/scalr/go-scalr"
+
+	"github.com/scalr/terraform-provider-scalr/internal/framework"
+	"github.com/scalr/terraform-provider-scalr/internal/framework/defaults"
+	"github.com/scalr/terraform-provider-scalr/internal/framework/validation"
+)
+
+// Compile-time interface checks
+var (
+	_ resource.Resource                     = &tagResource{}
+	_ resource.ResourceWithConfigure        = &tagResource{}
+	_ resource.ResourceWithConfigValidators = &tagResource{}
+	_ resource.ResourceWithImportState      = &tagResource{}
+)
+
+func newTagResource() resource.Resource {
+	return &tagResource{}
+}
+
+// tagResource defines the resource implementation.
+type tagResource struct {
+	framework.ResourceWithScalrClient
+}
+
+// tagResourceModel describes the resource data model.
+type tagResourceModel struct {
+	Id        types.String `tfsdk:"id"`
+	Name      types.String `tfsdk:"name"`
+	AccountID types.String `tfsdk:"account_id"`
+}
+
+func (r *tagResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_tag"
+}
+
+func (r *tagResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Manages the state of tags in Scalr.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The ID of this resource.",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Name of the tag.",
+				Required:            true,
+				Validators: []validator.String{
+					validation.StringIsNotWhiteSpace(),
+				},
+			},
+			"account_id": schema.StringAttribute{
+				MarkdownDescription: "ID of the account, in the format `acc-<RANDOM STRING>`.",
+				Optional:            true,
+				Computed:            true,
+				Default:             defaults.AccountIDRequired(),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+func (r *tagResource) ConfigValidators(_ context.Context) []resource.ConfigValidator {
+	return []resource.ConfigValidator{}
+}
+
+func (r *tagResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan tagResourceModel
```

Comment:
@petroprotsakh NIT: strange name for var `plan`

Reply:
well this variable actually contains a result of terraform plan 🤔